mutiny-core/src/storage.rs

```diff
@@ -67,12 +67,22 @@ pub trait MutinyStorage: Clone + Sized + 'static {
     fn cipher(&self) -> Option<Cipher>;
 
     /// Set a value in the storage, the value will already be encrypted if needed
-    fn set<T>(&self, key: impl AsRef<str>, value: T) -> Result<(), MutinyError>
+    fn set<T>(
+        &self,
+        key: impl AsRef<str>,
+        value: T,
+        version: Option<u32>,
+    ) -> Result<(), MutinyError>
```

Comment:
Doesn't have to be part of this, but are we really calling both `set` and `set_data` in different places? Might want to refactor this later or make the names / comments more clear when to use what.

Reply:
Yeah, `set_data` is what we use internally and `set` is what the user implements